src/api/docgen/render/markdown.ts

```diff
@@ -0,0 +1,193 @@
+import * as reflect from "jsii-reflect";
+
+/**
+ * Options for defining a markdown header.
+ */
+export interface MarkdownHeaderOptions {
+  /**
+   * Title to be displayed.
+   */
+  readonly title?: string;
+
+  /**
+   * Superscript.
+   *
+   * @default - No superscript
+   */
+  readonly sup?: string;
+
+  /**
+   * Preformat the header.
+   *
+   * @default false
+   */
+  readonly pre?: boolean;
+
+  /**
+   * Strikethough the title.
+   *
+   * @default false
+   */
+  readonly strike?: boolean;
+}
+
+/**
+ * Options for defining a markdown element.
+ */
+export interface MarkdownOptions {
+  /**
+   * Markdown header.
+   *
+   * @default - No header.
+   */
+  readonly header?: MarkdownHeaderOptions;
+
+  /**
+   * Id of the element.
+   *
+   * @default - The title will be used.
+   */
+  readonly id?: string;
+}
+
+/**
+ * Markdown element.
+ */
+export class Markdown {
+  /**
+   * An empty markdown element.
+   */
+  public static readonly EMPTY = new Markdown();
+
+  /**
+   * Sanitize markdown reserved characters from external input.
+   */
+  public static sanitize(line: string): string {
+    let sanitized = line.trim();
+
+    if (line.startsWith("-")) {
+      sanitized = sanitized.substring(1, line.length).trim();
+    }
+
+    return sanitized;
+  }
+
+  public static bold(text: string): string {
+    return `**${text}**`;
+  }
+
+  public static pre(text: string): string {
+    return `\`${text}\``;
+  }
+
+  public static italic(text: string) {
+    return `*${text}*`;
+  }
+
+  private readonly _lines = new Array<string>();
+  private readonly _sections = new Array<Markdown>();
+
+  private readonly id?: string;
+  private readonly header?: string;
+
+  constructor(private readonly options: MarkdownOptions = {}) {
+    this.id = options.id ?? options.header?.title;
+    this.header = this.formatHeader();
+  }
+
+  /**
+   * Render a `jsii-reflect.Docs` element into the markdown.
+   */
+  public docs(docs: reflect.Docs) {
+    if (docs.summary) {
+      this.lines(Markdown.sanitize(docs.summary));
+      this.lines("");
+    }
+    if (docs.remarks) {
+      this.lines(Markdown.sanitize(docs.remarks));
+      this.lines("");
```

Comment:
I saw in the example you linked that if "remarks" include markdown formatted text (eg bullets), then the output looks odd.

Wondering maybe all remarks should be under a block quote so if they are formatted it will look ok

Reply:
I actually started out this way and had some formatting issues as well. Also there are other things in quote right now (`see`, `link`) so it will cause a quote overload. 

Lets iterate on this later on.

Reply:
Okay. I'd argue that links and see can  also qualify as remarks